### PR TITLE
Allow packager to be opened in specific terminal on Linux and Mac OS X

### PR DESCRIPTION
**Pull request by smvv**
Thanks for submitting a PR! Please read these instructions carefully:

- [ ] Explain the **motivation** for making this change.
- [ ] Provide a **test plan** demonstrating that the code is solid.
- [ ] Match the **code formatting** of the rest of the codebase.
- [ ] Target the `master` branch, NOT a "stable" branch.

## Motivation (required)

What existing problem does the pull request solve?

Currently, it is not possible to spawn the packager in a specific terminal on Linux and Mac OS X.

For example, after applying this patch, starting the packager in a new xfce terminal on Linux can be
done using:

    react-native run-android --terminal /usr/bin/xfce4-terminal

When the command is ran a second time, and the terminal is still running, the command will not spawn a new terminal for the packager.

The option 'open' is renamed to 'terminal' for consistency. Note that the option 'open' was never exposed to the CLI though.

## Test Plan (required)

A good test plan has the exact commands you ran and their output, provides screenshots or videos if the pull request changes UI or updates the website. See [What is a Test Plan?][1] to learn more.  

If you have added code that should be tested, add tests.

See command above.

## Next Steps

Sign the [CLA][2], if you haven't already.

Small pull requests are much easier to review and more likely to get merged. Make sure the PR does only one thing, otherwise please split it.

Make sure all **tests pass** on both [Travis][3] and [Circle CI][4]. PRs that break tests are unlikely to be merged.

For more info, see the ["Pull Requests"][5] section of our "Contributing" guidelines.

[1]: https://medium.com/@martinkonicek/what-is-a-test-plan-8bfc840ec171#.y9lcuqqi9
[2]: https://code.facebook.com/cla
[3]: https://travis-ci.org/facebook/react-native
[4]: http://circleci.com/gh/facebook/react-native
[5]: https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#pull-requests

### CODE REVIEW

**Comment by grabbou** [comment]
> Note that the option 'open' was never exposed to the CLI though.

Nice finding!

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -250,15 +250,15 @@ function startServerInNewWindow() {
   const procConfig = {cwd: packagerDir};
 
   if (process.platform === 'darwin') {
-    if (yargV.open) {
```
Comment (grabbou) [inline local-cli/runAndroid/runAndroid.js]: Can we use `env` variable for consistency with other places where we already use it? 

Example: https://github.com/facebook/react-native/blob/dfaa03e1cb12b349152ca10f997ac198ab995d1e/local-cli/server/util/launchEditor.js#L84

`REACT_TERMINAL` or something might be a good idea for future scripts

**Comment by grabbou** [comment]
Overall looks good! Thanks for submitting. Just left a quick question (to consider). Let me know what you think :)

**Comment by hramos** [comment]
Still interested in landing this?

**Comment by smvv** [comment]
@hramos I've updated the commit with the requested changes, and rebased on top of master.

**Pull request closed without merging**